### DISCUSSION TITLE
terminations_b not showing as cabled.

### DISCUSSION THREAD

**Discussion by AaronJamesMyers** [netbox-community/netbox#13211] [Help Wanted!] [answered]
Netbox v3.4.10

Im trying to write a script that creates a connection between 2 interfaces within Netbox custom scripts using the following code:

a_term = []
a_term.append(cdu_interface)
            
b_term = []
b_term.append(data['uplink_interface'])     

_uplink_cable = Cable(
a_terminations=a_term,
b_terminations=b_term
    )
_uplink_cable.save()

The Cable seems to be created fine, and looks fine on interface A.  But on interface B, it is not showing a connection.  In the object for Interface B,  cable null.

When i look at the cable object, it shows the expected interface in terminations_b, but again, cable: null.

I tried looking up the termination B object in its own variable, and using that to append to the b_term list, but got the same result.

Any suggestions here??

**Comment by jeremystretch** [comment]
The terminations list must contain an Interface object(s), not just an ID or other raw data.

**Reply by AaronJamesMyers** [reply]
I added this into my code:
uplink_interface = Interface.objects.get(name = data['uplink_interface'].name, device = data['uplink_device'].id)

and used "uplink_interface" as  my  b_term.

a_term = []
a_term.append(cdu_interface)
            
b_term = []\
b_term.append(uplink_interface)     

_uplink_cable = Cable(
       a_terminations=a_term,
       b_terminations=b_term
           )
_uplink_cable.save()

I get the same result.  When i look at my cable, and its termination points, it shows the proper interfaces for termination A and termination B.  But nothing on the interface object at termination B

**Reply by abhi1693** [reply]
Try running full_clean() before you invoke save

**Reply by AaronJamesMyers** [reply]
a_term = []
            a_term.append(cdu_interface)
            
            b_term = []
            b_term.append(uplink_interface)     
            #b_term.append(uplink_interface)     

            _uplink_cable = Cable(
                a_terminations=a_term,
                b_terminations=b_term
               
                )
            _uplink_cable.full_clean()
            _uplink_cable.save()


same results.

here is what the newly created cable looks like:

"count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 19393,
      "url": "http://localhost:5000/api/dcim/cables/19393/",
      "display": "#19393",
      "type": "",
      "a_terminations": [
        {
          "object_type": "dcim.interface",
          "object_id": 408880,
          "object": {
            "id": 408880,
            "url": "http://localhost:5000/api/dcim/interfaces/408880/",
            "display": "Interface",
            "device": {
              "id": 16271,
              "url": "http://localhost:5000/api/dcim/devices/16271/",
              "display": "TESTSK01Q92 (55555)",
              "name": "TESTSK01Q92"
            },
            "name": "Interface",
            "cable": 19393,
            "_occupied": true
          }
        }
      ],
      "b_terminations": [
        {
          "object_type": "dcim.interface",
          "object_id": 179163,
          "object": {
            "id": 179163,
            "url": "http://localhost:5000/api/dcim/interfaces/179163/",
            "display": "SFP-3 (1/1/3)",
            "device": {
              "id": 5392,
              "url": "http://localhost:5000/api/dcim/devices/5392/",
              "display": "DVSNSK03O02",
              "name": "DVSNSK03O02"
            },
            "name": "SFP-3",
            "cable": null,
            "_occupied": false
          }
        }
      ],
      "status": {
        "value": "connected",
        "label": "Connected"
      },
      "tenant": null,
      "label": "",
      "color": "",
      "length": null,
      "length_unit": null,
      "description": "",
      "comments": "",
      "tags": [],
      "custom_fields": {},
      "created": "2023-07-19T15:01:59.255434Z",
      "last_updated": "2023-07-19T15:01:59.255447Z"


Also thanks for any suggestions here, I really do appreciate it. I realize everyone's time is important.

**Comment by AaronJamesMyers** [comment, marked as answer]
This was my bad guys.

I had pulled the object prior to creating the cable.  then saved that same object AFTER creating the cable. 

100% my bad.

Thank again for the suggestions
